### PR TITLE
APIv2 ExecStart (Attached Only)

### PR DESCRIPTION
**Pull request by mheon**
This implements ExecStart for APIv2. It presently only supports attached, streaming exec.

Not really tested yet.

Missing:
- APIv2 bindings for exec start and resize
- Bindings tests for exec
- Exec reaper for APIv2 server to reap dead exec sessions from running containers

What's not going to be in here:
- Podmanv2 tunnel client implementation: requires writing a frontend for our attach code, which is definitely out of scope

### CODE REVIEW

**Comment by mheon** [comment]
@marusak FYI. I suspect this isn't really working yet, but it's pretty close.

**Comment by mheon** [comment]
Update: session starts, seems to display STDOUT correctly. Still does not correctly handle STDIN, but seems a lot closer than it was.

**Comment by mheon** [comment]
TODO: Failing to sync exec session status as container was completely removed should not be fatal

**Comment by mheon** [comment]
```ERRO[0021] Received unexpected attach type +3, discarding 130 bytes ```

Hmmm.
@haircommander Any reason for Conmon to be sending us a 3 header in a terminal attach?

**Comment by haircommander** [comment]
> `ERRO[0021] Received unexpected attach type +3, discarding 130 bytes `
> 
> Hmmm.
> @haircommander Any reason for Conmon to be sending us a 3 header in a terminal attach?

uh not to my knowledge

**Comment by rhatdan** [comment]
@mheon Whats going on with this one?

**Comment by mheon** [comment]
Picking it back up on Monday

**Comment by mheon** [comment]
Should be working now. @marusak If you're interested in testing, would be appreciated.

**Comment by mheon** [comment]
Stripping WIP.

**Comment by mheon** [comment]
This should be ready now. @rhatdan @vrothberg @baude @giuseppe @TomSweeneyRedHat @jwhonce PTAL

**Comment by baude** [comment]
lgtm needs a peek from @jwhonce

File: libpod/oci_conmon_linux.go [hunk 1]
```diff
@@ -1712,6 +1644,7 @@ func httpAttachTerminalCopy(container *net.UnixConn, http *bufio.ReadWriter, cid
 				// Do nothing
 			default:
 				logrus.Errorf("Received unexpected attach type %+d, discarding %d bytes", buf[0], numR)
+				logrus.Debugf("String is %s", string(buf[1:numR]))
```
Comment (jwhonce) [inline libpod/oci_conmon_linux.go:1647]: This debug can be HUGE depending on the amount read.  That is why I deleted it the first time around.

File: pkg/api/handlers/compat/exec.go [hunk 2]
```diff
@@ -104,4 +106,115 @@ func ExecInspectHandler(w http.ResponseWriter, r *http.Request) {
...
+	if !utils.IsLibpodRequest(r) && inspectOut.CanRemove {
+		logrus.Infof("Pruning stale exec session %s from container %s", sessionID, sessionCtr.ID())
+		if err := sessionCtr.ExecRemove(sessionID, false); err != nil && errors.Cause(err) != define.ErrNoSuchExecSession {
+			logrus.Errorf("Error removing stale exec session %s from container %s: %v", sessionID, sessionCtr.ID(), err)
+		}
+	}
+}
+
+// ExecResizeHandler resizes a given exec session's TTY.
+func ExecResizeHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (jwhonce) [inline pkg/api/handlers/compat/exec.go:121]: I have this functionality in my PR as well...

File: libpod/oci_conmon_linux.go (hunk 1 above)
Comment (mheon) [reply libpod/oci_conmon_linux.go:1647]: Good point. I'll drop this entirely

File: pkg/api/handlers/compat/exec.go (hunk 2 above)
Comment (mheon) [reply pkg/api/handlers/compat/exec.go:121]: For Exec?

Comment (mheon) [reply pkg/api/handlers/compat/exec.go]: Fixed

File: libpod/oci_conmon_linux.go (hunk 1 above)
Comment (mheon) [reply libpod/oci_conmon_linux.go]: Fixed

**Comment by mheon** [comment]
Comments addressed

**Pull request merged**